test/integration/puma/plugin_test.rb

```diff
@@ -7,17 +7,18 @@ class PumaPluginTest < ActiveSupport::TestCase
   setup do
     FileUtils.mkdir_p Rails.root.join("tmp", "pids")
 
-    cmd = %w[
-      bundle exec puma
-        -b tcp://127.0.0.1:9222
-        -C test/dummy/config/puma.rb
-        --dir test/dummy
-        -s
-        config.ru
-    ]
-
-    @pid = fork do
-      exec(*cmd)
+    Dir.chdir('test/dummy') do
```

Comment:
I had to change the dir here this way instead of using `--dir` option because when you trigger puma's restart it can't find the app anymore since current dir is already `test/dummy` and it's trying to use `test/dummy` in addition.